src/components/Form.js

```diff
@@ -149,12 +150,30 @@ export default class Form extends Component {
     }
   };
 
+  getUsedFormData = event => {
+    const fields = event.target.elements;
+    const inputNames = Object.keys(fields).reduce((filtered, i) => {
+      const fieldName = fields[i].name;
+      if (fieldName) {
+        filtered.push(fieldName);
+      }
+      return filtered;
+    }, []);
+
+    return _pick(this.state.formData, inputNames);
+  };
+
   onSubmit = event => {
     event.preventDefault();
     event.persist();
+    let newFormData = this.state.formData;
+
+    if (this.props.omitUnusedData) {
+      newFormData = this.getUsedFormData(event, this.state.formData);
```

Comment:
@travisdahl Can you explain why you decided to call `getUsedFormData` only on `onSubmit` and not during `onChange`?

Reply:
Hello! I worked on this with @travisdahl so I think I can answer. We decided to only add it to onSubmit because if we omitted the data in onChange then you would lose the functionality of hidden/dependent fields being populated with data when being shown.

Reply:
Yeah @epicfaace , we went back and forth about this, on whether to always omit it or only on submit.  In the issue thread people were arguing they dont want to lose data while showing, hiding dependent fields which is understandable, so we thought that when you "submit" a form that is a more final action, then you have applied your changes, saved, etc. But now that im saying it aloud, I guess it really depends on how you use the component. We do some stuff on change but dont save to our redux store until submit. Other may only use `onChange` that would send data to the store on each change. We can change it to be `onChange` instead but then you will lose data while hiding showing fields on dependencies. Not to sure what the right expectation would be, would love to hear your opinion.

Reply:
I'd like for it to happen on `onChange` if possible. I don't quite get what you mean when you say "you will lose data while hiding showing fields on dependencies" -- could you give an example of this situation?

Reply:
lets use the example in the docs...

If I enter a credit card number, then the billing address field would be shown. If I then entered an address, then cleared my credit card number. The address field would be hidden (and data discarded). The if I entered a new credit card number, I would have to re-enter my address. Which on one hand, is what i would expect if I set omit extra data to true BUT its kind of a bummer for the user who has to re-enter the address again.

Reply:
How about this?
- if `liveValidate` is true, then we perform the extra data omission during `onChange`
- if `liveValidate` is false, then we perform the extra data omission during `onSubmit`

Reply:
That sounds good, I was just about to push my onChange changes, so I'll modify it to work like this.